### ISSUE TITLE
SiteWrapper: Nav links in account dropdown

### ISSUE THREAD

**Issue by meiremans** [tabler/tabler-react#424]
Hi
 I cannot seem to figure this one out.
When i use the regular navbar:
`
const navBarItems: Array<navItem> = [
  ...
  { icon: "user", value: "Profile",  to: "/profile", LinkComponent: withRouter(NavLink) , useExact: true  }
];`

This works as intended and the profile page loads with my Redux store

However if i want the profile link where it is supposed to be:
`let accountDropdownProps = {
  avatarURL: "./demo/faces/female/25.jpg",
  name: "Jane Pearson",
  description: "Administrator",
  options: [
     ...
    { icon: "user", value: "Profile",  to: "/profile", LinkComponent: withRouter(NavLink) , useExact: true  },
  ],
};
`
The profile page will be loaded, but my store won't be persisted in this route change.

I have the feeling this is a bug in the AccountDropdown element or am I missing something?

**Comment by Levino** [comment]
It is just not supported. So not a bug but a feature. I have to say is way too complicated here. I mean: I am seriously not able to implement a logout functionality in a proper way:
- For the `onClick` function I need a history object to push to but for the history object I need to wrap the component factory function in `withRouter` which leads to attempt two:
- I pass in a custom component that I wrap with `withRouter` as a `rootComponent` property. Now on click I can logout or navigate client side to `/logout` but the dropdown menu stays open because its local state is not changed.
- I just add a `to: '/logout'`. That will then rerender my whole app, which is just annoying.

So long story short is that this library is not really working well. I will replace it in my code.

**Comment by jonthomp** [comment]
@meiremans as @Levino suggests, this is a restriction in how we built version 1. If you are still using Tabler, you might be interested to know the WIP v2 solves these issues and gives you whole lot more freedom. You can check out the progress on [branch](https://github.com/tabler/tabler-react/tree/version-2). I hope to release an alpha version in the next week.
